### DISCUSSION TITLE
Add a column(target)

### DISCUSSION THREAD

**Discussion by ozi42** [biolab/orange3#5562] [Q&A]
Hello,

i want to change the domain of a DataTable by adding a target-column.

The problem is : they added a new column, but not the values.

---------------------------------------
code:

data=copy.deepcopy(in_data)
print(data)
 
<< [[1.0, 2, 5, 3, 3, 9],
<< [1.0, 5, 10, 2, 4, 9],
<< [3.0, 5, 2, 5, 1, 1],
<< [0.9, 4, 5, 8, 8, 5]]

print(data.domain)

<<[ Jan, Feb, März, Apr, Mai, Juni]

#now add the new target

target_values=('L','kk','24','5')
a=DiscreteVariable(name="KOOO",values=(target_values))

#now add it
data.domain.__init__(attributes=data.domain.attributes,class_vars=a)

print(daten)

<<[[1.0, 2, 5, 3, 3, 9 | ],
<< [1.0, 5, 10, 2, 4, 9 | ],
<<[3.0, 5, 2, 5, 1, 1 | ],
<<[0.9, 4, 5, 8, 8, 5 | ]]

---------------------------------------

**Comment by markotoplak** [comment]
`values` is a list of available values for a discrete variable.

You still need to create a new table with the new domain. It would look approximately like:

```
new_domain = Ornage.data.Domain(attributes=data.domain.attributes,class_vars=a)
new_data = data.transform(new_domain)
```

Now you would get a new table with still empty values. But then you could edit new_data.Y to fill the values in, with `data.Y[0] = value_index`.

Please read the docs at https://orange3.readthedocs.io/projects/orange-data-mining-library/en/latest/tutorial/data.html
